### PR TITLE
[#11572] Notification Feature - Notification Banner

### PR DESCRIPTION
**Pull request by zhaojj2209**
Part of #11572 

Adds notification banner that shows on instructor/student pages for logged in users.

Example notif:

https://user-images.githubusercontent.com/54243224/163118903-9db2ccb7-cb4e-4999-8481-2a353dbba48c.mov

Also included is a bug fix where notifications marked as read by admins will not be shown on the admin notifications page; the fix has been bundled here instead of a separate PR given time constraints.

Future extensions to notif banner UI not included in this PR:
- Adding ability to toggle between multiple unread notifs
- Truncating notification text with a "read more" link that redirects users to user notifications page (currently, notification text is displayed in full with option to scroll)

### CODE REVIEW

**Comment by zhaojj2209** [comment]
@damithc Thank you for the suggestions! Changes have been made, updates to UI are as follows:
Notif banner, user notifs page
Screenshot 2022-04-13 at 17 57 17
Admin notif edit form
Screenshot 2022-04-13 at 17 57 32
Admin notifs table
Screenshot 2022-04-13 at 17 58 01
Marking notif as read

https://user-images.githubusercontent.com/54243224/163158972-cb1abc5e-b82e-43e7-bf38-9f4d1de259d5.mov

**Comment by damithc** [comment]
> @damithc Thank you for the suggestions! Changes have been made, updates to UI are as follows:

Looks better now 👍 

The `secondary` style button is certainly more neutral but not sure it's a good look though. Perhaps try outline style?
image

Furthermore, padding around the button might need tweaking.
image

It's a bit nitpicky but given notifications could the the first thing a new user will see when they login, we should make it look good :-)

**Review by jianhandev — COMMENTED** [review]
UI-wise looks quite much better after the change to lighter colours. Just some personal thoughts for future consideration: 
* Increasing the roundness of the borders
* Adding some padding especially top and bottom
* Adding some margin to show some white space around the banner

A mockup to show what it might look like after the changes:

Screenshot 2022-04-13 at 8 37 46 PM

File: src/main/java/teammates/ui/webapi/GetNotificationsAction.java [hunk 1]
```diff
@@ -79,10 +80,6 @@ public JsonResult execute() {
                 .filter(n -> !readNotifications.contains(n.getNotificationId()))
                 .collect(Collectors.toList());
 
-        if (userInfo.isAdmin) {
-            return new JsonResult(new NotificationsData(notificationAttributes));
-        }
-
```
[the remark this answers is not in the document]
Comment (zhaojj2209) [reply src/main/java/teammates/ui/webapi/GetNotificationsAction.java:85]: Got it, will add this check back!

**Comment by zhaojj2209** [comment]
> The `secondary` style button is certainly more neutral but not sure it's a good look though. Perhaps try outline style?

@damithc Outline style buttons upon hover look exactly the same as normal buttons, so it probably wouldn't help much. I've thought of an alternative solution, which is to change the style of the button along with the notification style:
Screenshot 2022-04-13 at 20 57 10
Screenshot 2022-04-13 at 20 57 38

**Review by t-cheepeng — APPROVED** [review]
Other than the comments by jianhan, looks good. UI looks good as well:+1:

**Comment by zhaojj2209** [comment]
@ziqing26 @jianhandev To allow for the notif padding to easily change according to the padding of the main content, I've moved the notif banner within the main content container:
Screenshot 2022-04-14 at 01 47 16

Should I retain the notif's sticky behaviour, or have it scroll with the page?

**Comment by ziqing26** [comment]
> @ziqing26 @jianhandev To allow for the notif padding to easily change according to the padding of the main content, I've moved the notif banner within the main content container: Screenshot 2022-04-14 at 01 47 16
> 
> Should I retain the notif's sticky behaviour, or have it scroll with the page?

In my opinion, leaving it as sticky is fine, scroll with the page might make it look too bulky.

**Comment by ziqing26** [comment]
> Screenshot 2022-04-14 at 01 47 16

Moreover, is it okay to add more margin at right and top for the mark as read button? It might be better to align to left with the text body instead of the scroll bar.

Meanwhile, it might be better to reduce the spacing below the button as well, this is just a suggestion for discussion. Thanks Jingjing!

**Review by jianhandev — APPROVED** [review]
LGTM. Thanks for the UI changes! It looks great now 👍

**Comment by damithc** [comment]
I'm OK with the current version as well.

**Review by ziqing26 — APPROVED** [review]
LGTM! Thank you Jingjing

**Review by NicolasCwy — APPROVED** [review]
Overall looks good to me! Just a nit that its a bit hard to differentiate the new `light` style on the admin side since there isn't a border, as compared to the previous style. I think its not important for now since its an admin side feature, and the style is still clear on the user side

Current Light Style
image

Previous Light Style
image

**Comment by zhaojj2209** [comment]
> Overall looks good to me! Just a nit that its a bit hard to differentiate the new `light` style on the admin side since there isn't a border, as compared to the previous style. I think its not important for now since its an admin side feature, and the style is still clear on the user side

Thanks for the catch! Since this affects only admin side and only one style, I agree that we can leave this for a future PR.

**Review by jianhandev — APPROVED** [review]

**Pull request merged**